public/main.js

```diff
@@ -83,7 +90,13 @@ function reloadTodoList() {
                 });
             };
             listItem.appendChild(completeButton);
-            todoList.appendChild(listItem);
+            if (displayTodos === "all") {
```

Comment:
I noticed 3 things here. First, for the logic you have added here, since you are adding the item in all three cases, you can do something like 
```
var displayItem = displayTodos === "all" 
    || (displayTodos === "completed" && todo.isComplete) 
    || (displayTodos === "active" && !todo.isComplete);
if (displayItem) {
    todoList.appendChild(listItem);
}
```

Second it looks like your code for creating the listItem is slightly in the wrong order. You are creating the item, then checking whether it should be added. Instead, check whether the item should be used, then create and add it. 

Last, your reloadTodoList function has become quite big. To shorten it, split out the callback for getTodoList and call it createTodoList.